User, Media, Password, and other Fundamental Concepts

HenryAsa opened the pull request:
Added a variety of different fundamental concepts that are necessary for the development of the rest of the app.

HenryAsa committed:
Added `user` and `media` concepts

HenryAsa committed:
Added Firebase

HenryAsa committed:
Updated some styling

HenryAsa committed:
Added `media`

HenryAsa committed:
Updated `user`

HenryAsa committed:
Updated firebase

HenryAsa committed:
Committing changes

HenryAsa committed:
Fixed `Firebase` bug

HenryAsa committed:
Looks better now

HenryAsa committed:
Simplified the user profile feature

HenryAsa committed:
Added the Profile Photo to the icon bar

DragonStorm25 committed:
Merge branch 'main' into base-concepts

HenryAsa committed:
Minor change

DragonStorm25 reviewed (changes requested):
I really don't like the error throwing within isCreator and isNotCreator. Also, you should only need one of those since they're inverses of each other. Please move the error throwing outside of isCreator/isNotCreator and get rid of one of them.

Personal user information (like the password!) is printed into the console, that does not seem secure at all.

HenryAsa committed:
Removed unsecure console.log print statements as per Jonatans request

HenryAsa committed:
Changed default error handling of `isCreator` and `isNotCreator`

HenryAsa commented:
> I really don't like the error throwing within isCreator and isNotCreator. Also, you should only need one of those since they're inverses of each other. Please move the error throwing outside of isCreator/isNotCreator and get rid of one of them.
> 
> Personal user information (like the password!) is printed into the console, that does not seem secure at all.

Thanks @DragonStorm25, great feedback.  I removed those `console.log()` statements you were talking about (great catch).  I also updated the default behavior of `isCreator` and `isNotCreator` so that they **do not throw errors** unless the `throw_error` attribute is set to `true` (by default it is set to `false`).  I know it might seem redundant to have both `isCreator()` and `isNotCreator()`, but it's super useful when I _want_ to throw errors.  Rather than having an `if` statement check whether the output of these functions is true or false, and then writing the errors to be thrown after that, I can just set the `throw_error = true` in the function call and this is all handled automatically.  I'm gonna keep it because it's quite helpful for synchronizations that we will write in `routes.ts`.

Thanks for reviewing everything!
